app/models/organization_resolver.rb

```diff
@@ -48,6 +49,25 @@ def organization_tag_names
     @organization&.tags || []
   end
 
+  def entities
+    [
+      @organization,
+      @email,
+      @location,
+      @organization_name,
+      @phone_number,
+      @website
+    ]
+  end
+
+  def source_names
+    entities.reduce([]) do |memo, entity|
+      next entity if entity.nil?
+      name = entity.versions.last.actor.import.source.name
+      memo.include?(name) ? memo : memo << name
+    end
```

Comment:
My thinking is that with `reduce` I only loop once, instead of multiple times. Admittedly, not a big deal with an array with fewer than ten items. If it's a better read for people, I'm happy to do it the way you suggest.